### ISSUE TITLE
GitHub support for xtlang

### ISSUE THREAD

**Issue by benswift** [digego/extempore#69]
Currently, GitHub doesn’t recognise xtlang, and so syntax highlighting doesn’t work for .xtm files or gists.

Adding xtlang support to GitHub would involve writing a [linguist](https://github.com/github/linguist) parser for xtlang (inheriting from the existing Scheme parser would probably be a good place to start).

**Comment by mrmagooey** [comment]
As per [the Linguist Contributing Guide] (https://github.com/github/linguist/blob/master/CONTRIBUTING.md) we are still several hundred github repositories away from being able to contribute to Linguist and have xtlang recognised within Github with syntax highlighting. A search for extempore and xtlang within github only comes up with < 40 repositories currently.

Given that this is not an issue that can be addressed from within the extempore project, I suggest the issue is closed.

**Comment by jasonlevine** [comment]
But we can still make and share the extension with Extempore users in the
meantime, right?

On Mon, Dec 7, 2015 at 2:51 PM, mrmagooey <notifications@github.com> wrote:

> As per the Linguist Contributing Guide
> <https://github.com/github/linguist/blob/master/CONTRIBUTING.md> we are
> still several hundred github repositories away from being able to
> contribute to Linguist and have xtlang recognised within Github with syntax
> highlighting. A search for extempore and xtlang within github only comes up
> with < 40 repositories currently.
>
> Given that this is not an issue that can be addressed from within the
> extempore project, I suggest the issue is closed.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/digego/extempore/issues/69#issuecomment-162429086>.
>



-- 
Jason Levine
new media performer + creative coder
*https://www.behance.net/jasonlevine <https://www.behance.net/jasonlevine>*

**Comment by mrmagooey** [comment]
Is there a method of selectively distributing Linguist extensions? I got the impression we needed to contribute to the core Linguist library if we want syntax highlighting in Github.

**Comment by snowquiet** [comment]
I found a workaround to get syntax highlighting for .xtm files on github without any upstream changes to linguist. Linquist already knows about scheme, and it supports language overrides with emacs file variables. So, adding this to the top of a .xtm file will cause linguist to use scheme syntax highlighting:
```
;; -*- mode: scheme -*-
```
It works for gists and regular files in github:
![fmsynth-xtm-with-modeline](https://user-images.githubusercontent.com/44050863/51579703-97efbc80-1e87-11e9-9ec7-720e6b93b2ef.png)

According to the [linguist manual override documentation](https://github.com/github/linguist#overrides) It might be possible to get the same effect for all .xtm files in a repository by creating a `.gitattributes` file like this:
```
*.xtm linguist-language=Scheme
```
But, when I tested that approach in my fork, it didn't seem to work. The comments in [linguist issue 1792](https://github.com/github/linguist/issues/1792#issuecomment-286379822) suggest that it ought to work but that caching on the github end can be an issue. Perhaps I didn't wait long enough. Maybe it only works for commits in the master branch.

**Comment by benswift** [comment]
Yep, that's a good catch - thanks  @snowquiet . I'll check that it doesn't break anything else and make the change.
